File: 0560_subarraySumEqualsK/main.py
from typing import List
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        res = 0
        curSum = 0
        prefixSum = {0 : 1}
        
        for num in nums:
            curSum += num
            diff = curSum - k

            res += prefixSum.get(diff, 0)
            prefixSum[curSum] = prefixSum.get(curSum, 0) + 1

        return res

File: 0560_subarraySumEqualsK/main.cpp
#include <vector>
#include <unordered_map>

class Solution {
public:
    int subarraySum(std::vector<int>& nums, int k) {
        std::unordered_map<int,int> prefixSum;
        int curSum = 0;
        int res = 0;
        prefixSum[curSum] = 1;
        for(auto it:nums){
            curSum += it;
            int diff = curSum - k;
            if(prefixSum.find(diff) != prefixSum.end()){
                res += prefixSum[diff];
            }
            ++prefixSum[curSum];
        }
        return res;        
    }
};